pkg/kafka/config.go

```diff
@@ -92,9 +110,21 @@ func (cfg *Config) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 }
 
 func (cfg *Config) Validate() error {
```

Comment:
I believe you currently allow mixing and matching of the old and new configurations, for example:

```
kafka_config:
    address: localhost:9092
    reader:
        client_id: test
```

Reply:
Not sure if that was intentional?